### PR TITLE
include m3 dataset

### PR DESCRIPTION
**Pull request by vafl**
*Issue #, if available:*

*Description of changes:*
Include m3 dataset.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/dataset/repository/_m3.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+def generate_m3_dataset(dataset_path: Path, m3_freq: str):
+    from gluonts.dataset.repository.datasets import default_dataset_path
+
+    m3_xls_path = default_dataset_path / "M3C.xls"
+    if not os.path.exists(m3_xls_path):
+        raise RuntimeError(
+            f"The m3 data is available at https://forecasters.org/resources/time-series-data/m3-competition/ "
+            f"Please download the file and copy the files to this location: {m3_xls_path}"
+        )
```
Comment (lostella) [inline src/gluonts/dataset/repository/_m3.py:63]: Couldn't the original data be downloaded automatically?

Comment (lostella) [reply src/gluonts/dataset/repository/_m3.py:63]: This is what happens for M4 at least: https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/dataset/repository/_m4.py

Comment (vafl) [reply src/gluonts/dataset/repository/_m3.py:63]: The server is blocking automated requests. We can spoof the user agent, but that seems a bit too dodgy for including it in the package.

Comment (vafl) [reply src/gluonts/dataset/repository/_m3.py:63]: Ideally, we would host the file ourselves, but there is no licence on the website. It is however also included in this gpl3 package: https://github.com/robjhyndman/Mcomp/tree/master/data

**Review by lostella — APPROVED** [review]

**Pull request merged**